packages/cozy-dataproxy-lib/src/search/SearchEngine.ts

```diff
@@ -197,25 +208,83 @@ export class SearchEngine {
     this.subscribeDoctype(this.client, CONTACTS_DOCTYPE)
     this.subscribeDoctype(this.client, APPS_DOCTYPE)
 
+    const pouchLink = getPouchLink(this.client)
+    if (pouchLink) {
+      const sharedDrivesDoctypes = pouchLink.getSharedDriveDoctypes()
+      for (const sharedDrivesDoctype of sharedDrivesDoctypes) {
+        const driveId = sharedDrivesDoctype.split('-').pop()
+        if (driveId) {
+          this.addSharedDriveRealtime(driveId)
+        }
+      }
+    }
+
     if (this.isLocalSearch) {
       this.debouncedReplication()
     }
     // The document indexing should be performed once everything is setup
     await this.indexDocumentsAtInit()
   }
 
-  subscribeDoctype(client: CozyClient, doctype: string): void {
+  addSharedDrive(driveId: string): void {
+    this.addSharedDriveRealtime(driveId)
+    if (this.isLocalSearch) {
+      this.debouncedReplication()
```

Comment:
This is not directly related to this PR, but I think we should definitely add a way to start a replication per doctype